File: CSharp/LiveUpdating/QuoteInfo.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - Live Updating Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [QuoteInfo.cs]
 * 
 * This class represents a quote fetched information.
 * 
 * This file is part of the Xceed DataGrid for WPF product. The use 
 * and distribution of this Sample Code is subject to the terms 
 * and conditions referring to "Sample Code" that are specified in 
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 *
 */

using System;

namespace Xceed.Wpf.DataGrid.Samples.LiveUpdating
{
  public class QuoteInfo
  {
    public QuoteInfo( string symbol )
      : this( symbol, 0.0d, 0.0d, 0.0d, MarketSimulator.Popularities.Passive, MarketSimulator.Tendencies.Stable )
    {
    }

    internal QuoteInfo(
      string symbol,
      double previousClose,
      double openingValue,
      double lastTrade,
      MarketSimulator.Popularities popularity,
      MarketSimulator.Tendencies tendency )
    {
      m_symbol = symbol;
      m_previousClose = previousClose;
      m_openingValue = openingValue;
      m_lastTrade = lastTrade;
      m_tradeTime = "N/A";

      m_popularity = popularity;
      m_tendency = tendency;
    }

    #region Symbol PROPERTY

    public string Symbol
    {
      get { return m_symbol; }
    }

    private string m_symbol;

    #endregion Symbol PROPERTY

    #region PreviousClose PROPERTY

    public double PreviousClose
    {
      get { return m_previousClose; }
    }

    private double m_previousClose;

    #endregion PreviousClose PROPERTY

    #region OpeningValue PROPERTY

    public double OpeningValue
    {
      get { return m_openingValue; }
    }

    private double m_openingValue;

    #endregion OpeningValue PROPERTY

    #region LastTrade PROPERTY

    private double m_lastTrade;

    public double LastTrade
    {
      get { return m_lastTrade; }
    }

    #endregion LastTrade PROPERTY

    #region TradeTime PROPERTY

    private string m_tradeTime;

    public string TradeTime
    {
      get { return m_tradeTime; }
    }

    #endregion TradeTime PROPERTY

    #region INTERNAL METHODS

    internal void SetLastTrade( double lastTrade, DateTime tradeTime )
    {
      m_lastTrade = lastTrade;
      m_tradeTime = tradeTime.ToString( "HH:mm:ss:FF" );
    }

    #endregion INTERNAL METHODS

    #region INTERNAL PROPERTIES

    internal MarketSimulator.Popularities Popularity
    {
      get
      {
        return m_popularity;
      }
    }

    private MarketSimulator.Popularities m_popularity;

    internal MarketSimulator.Tendencies Tendency
    {
      get
      {
        return m_tendency;
      }
    }

    private MarketSimulator.Tendencies m_tendency;

    #endregion INTERNAL PROPERTIES
  }
}

File: Visual Basic .NET/LiveUpdating/QuoteInfo.vb
'
' * Xceed DataGrid for WPF - SAMPLE CODE - Live Updating Sample Application
' * Copyright (c) 2007-2024 Xceed Software Inc.
' * 
' * [QuoteInfo.vb]
' * 
' * This class represents a quote fetched information.
' * 
' * This file is part of the Xceed DataGrid for WPF product. The use 
' * and distribution of this Sample Code is subject to the terms 
' * and conditions referring to "Sample Code" that are specified in 
' * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
' *
' 


Imports System
Imports System.Collections.Generic

Namespace Xceed.Wpf.DataGrid.Samples.LiveUpdating
  Public Class QuoteInfo
    Public Sub New(ByVal symbol As String)
      Me.New(symbol, 0, 0, 0, MarketSimulator.Popularities.Passive, MarketSimulator.Tendencies.Stable)
    End Sub

    Friend Sub New(ByVal symbol As String, ByVal previousClose As Double, ByVal openingValue As Double, ByVal lastTrade As Double, ByVal popularity As MarketSimulator.Popularities, ByVal tendency As MarketSimulator.Tendencies)
      m_symbol = symbol
      m_previousClose = previousClose
      m_openingValue = openingValue
      m_lastTrade = lastTrade
      m_tradeTime = "N/A"

      m_popularity = popularity
      m_tendency = tendency
    End Sub

#Region "Symbol PROPERTY"

    Public ReadOnly Property Symbol() As String
      Get
        Return m_symbol
      End Get
    End Property

    Private m_symbol As String

#End Region

#Region "PreviousClose PROPERTY"

    Public ReadOnly Property PreviousClose() As Double
      Get
        Return m_previousClose
      End Get
    End Property

    Private m_previousClose As Double

#End Region

#Region "OpeningValue PROPERTY"

    Public ReadOnly Property OpeningValue() As Double
      Get
        Return m_openingValue
      End Get
    End Property

    Private m_openingValue As Double

#End Region

#Region "LastTrade PROPERTY"

    Private m_lastTrade As Double

    Public ReadOnly Property LastTrade() As Double
      Get
        Return m_lastTrade
      End Get
    End Property

#End Region

#Region "TradeTime PROPERTY"

    Private m_tradeTime As String

    Public ReadOnly Property TradeTime() As String
      Get
        Return m_tradeTime
      End Get
    End Property

#End Region

#Region "INTERNAL METHODS"

    Friend Sub SetLastTrade(ByVal lastTrade As Double, ByVal tradeTime As DateTime)
      m_lastTrade = lastTrade
      m_tradeTime = tradeTime.ToString("HH:mm:ss:FF")
    End Sub

#End Region

#Region "INTERNAL PROPERTIES"

    Friend ReadOnly Property Popularity() As MarketSimulator.Popularities
      Get
        Return m_popularity
      End Get
    End Property

    Private m_popularity As MarketSimulator.Popularities

    Friend ReadOnly Property Tendency() As MarketSimulator.Tendencies
      Get
        Return m_tendency
      End Get
    End Property

    Private m_tendency As MarketSimulator.Tendencies

#End Region
  End Class
End Namespace
